indico/htdocs/js/indico/jquery/categorypicker.js

```diff
@@ -0,0 +1,354 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    'use strict';
+
+    $.widget('indico.categorypicker', {
+        options: {
+            categoryId: 0,
+            height: null,
+            actionButtonText: $T.gettext("Select"),
+            openInDialog: false,
+            dialogTitle: $T.gettext("Select a category"),
+            selectLeafOnly: false, // Allow only the leaf categories to be selected
+            onAction: function() {}
+        },
+
+        // Cache for data of already visited categories
+        _cache: {},
+
+        _typeaheadTemplate:
+        '<form>\
+            <div class="typeahead__container">\
+                <div class="typeahead__field">\
+                    <span class="typeahead__query"></span>\
+                </div>\
+            </div>\
+        </form>',
+
+        _create: function() {
+            var self = this;
+            if (self.options.openInDialog) {
+                self._createInDialog();
+            } else {
+                self._createInline();
+            }
+        },
+
+        _createInline: function() {
+            var self = this;
+            self.element.addClass('category-picker');
+            self._createNavigator();
+            self._createSearchField();
+            self.goToCategory(self.options.categoryId);
+        },
+
+        _createInDialog: function() {
+            var self = this;
+            ajaxDialog({
+                title: self.options.dialogTitle,
+                content: $('<div>', {class: 'category-picker-wrapper'}).append($('<div>'))[0].outerHTML,
+                closeButton: '',
```

Comment:
- `undefined` or `false` --> no close button
- empty string or `true` --> default text
- non-empty string --> use that string for the button